### PR TITLE
Viewer: Adds copy to clipboard. Fixes #1125.

### LINKED ISSUE #1125: Viewer: copy to clipboard
We should have a copy to clipboard button in the nav that copies the JSON to the clipboard.

### PR DESCRIPTION
**Pull request by ebidel**
R: all

screen shot 2016-12-08 at 12 48 48 pm

BTW, the copy APIs are bizarre!!!!

Fixes https://github.com/GoogleChrome/lighthouse/issues/1125

### CODE REVIEW

File: lighthouse-viewer/app/src/lighthouse-report-viewer.js [hunk 1]
```diff
@@ -58,6 +61,23 @@ class LighthouseViewerReport {
     if (shareButton) {
       shareButton.addEventListener('click', this.onShare);
     }
+
+    const copyButton = document.querySelector('.js-copy');
+    if (copyButton) {
+      copyButton.addEventListener('click', this.onCopy);
+
+      document.addEventListener('copy', e => {
```
Comment (brendankenny) [inline lighthouse-viewer/app/src/lighthouse-report-viewer.js]: move this to it's own method?

also it's weird that there's `onCopy` which means that the copy button was clicked and this function, which is what would often be called `onCopy` :)

File: lighthouse-viewer/app/src/lighthouse-report-viewer.js [hunk 2]
```diff
@@ -164,6 +184,29 @@ class LighthouseViewerReport {
...
+        // Note: In Safari 10.0.1, execCommand('copy') returns true if there's
+        // a valid text selection on the page. See http://caniuse.com/#feat=clipboard.
+        const successful = document.execCommand('copy');
+        if (!successful) {
+          this._copyAttempt = false; // Prevent event handler from seeing this as a copy attempt.
+          logger.warn('Your browser does not support copy to clipboard.');
+        }
+      }
+    } catch (err) {
+      logger.log(err.message);
```
Comment (brendankenny) [inline lighthouse-viewer/app/src/lighthouse-report-viewer.js]: should set `this._copyAttempt = false` here?

Comment (ebidel) [inline lighthouse-viewer/app/src/lighthouse-report-viewer.js]: yea, can't hurt. Done.

File: lighthouse-viewer/app/src/lighthouse-report-viewer.js (hunk 1 above)
Comment (ebidel) [inline lighthouse-viewer/app/src/lighthouse-report-viewer.js]: done

File: lighthouse-viewer/app/src/lighthouse-report-viewer.js [hunk 3]
```diff
@@ -17,7 +17,7 @@
 
 'use strict';
 
-/* global ga */
+/* global window, ga */
```
Comment (ebidel) [inline lighthouse-viewer/app/src/lighthouse-report-viewer.js]: I got overly excited. Removed.

**Comment by ebidel** [comment]
PTAL

**Pull request merged**